tests/functional/inflight_close.cpp

```diff
@@ -7,309 +7,132 @@
-			OFINCCLCHECKGOTO(
-				initialize_buff((void *)send_buf[idx], send_size, buffer_type),
-				res, exit);
+		auto sComm = ctx.scomms.at(dev_idx);
+		auto rComm = ctx.rcomms.at(dev_idx);
 
-			OFINCCLCHECKGOTO(extNet->regMr((void *)sComm, (void *)send_buf[idx],
-							send_size, buffer_type, &mhandle[idx]),
-					 res, exit);
-			NCCL_OFI_TRACE(NCCL_NET,
-					"Successfully registered send memory for request %d of rank %d",
-					idx, rank);
-			while (req[idx] == NULL) {
-				OFINCCLCHECKGOTO(extNet->isend((void *)sComm, (void *)send_buf[idx],
-								send_size, tag, mhandle[idx],
-								(void **)&req[idx]),
-						 res, exit);
+		// Post operations
+		if (rank == 0) {
+			for (int i = 0; i < NUM_REQUESTS; i++) {
+				OFINCCLCHECK(allocate_buff(&buffers[i], DATA_SIZE, buffer_type));
+				OFINCCLCHECK(initialize_buff(buffers[i], DATA_SIZE, buffer_type));
+				OFINCCLCHECK(register_memory(ext_net, sComm, buffers[i], DATA_SIZE, buffer_type, &mhandles[i]));
+				OFINCCLCHECK(post_send(ext_net, sComm, buffers[i], DATA_SIZE, TAG, mhandles[i], &requests[i]));
 			}
-		}
-		NCCL_OFI_INFO(NCCL_NET, "Successfully sent %d requests to rank %d", NUM_REQUESTS,
-				peer_rank);
-	} else {
-
-		/* Receive NUM_REQUESTS from peer */
-		NCCL_OFI_INFO(NCCL_NET, "Rank %d posting %d receive buffers", rank,
-				NUM_REQUESTS);
-		for (int idx = 0; idx < NUM_REQUESTS; idx++) {
-			OFINCCLCHECKGOTO(
-				allocate_buff((void **)&recv_buf[idx], recv_size, buffer_type),
-				res, exit);
-			OFINCCLCHECKGOTO(extNet->regMr((void *)rComm, (void *)recv_buf[idx],
-							recv_size, buffer_type, &mhandle[idx]),
-					 res, exit);
-			NCCL_OFI_TRACE(NCCL_NET, "Successfully registered receive memory for request %d of rank %d", idx, rank);
-			while (req[idx] == NULL) {
-				OFINCCLCHECKGOTO(extNet->irecv((void *)rComm, nrecv,
-								(void **)&recv_buf[idx], sizes, tags,
-								&mhandle[idx], (void **)&req[idx]),
-						 res, exit);
+		} else {
+			for (int i = 0; i < NUM_REQUESTS; i++) {
+				OFINCCLCHECK(allocate_buff(&buffers[i], DATA_SIZE, buffer_type));
+				OFINCCLCHECK(register_memory(ext_net, rComm, buffers[i], DATA_SIZE, buffer_type, &mhandles[i]));
+
+				void* recv_bufs[] = {buffers[i]};
+				size_t sizes[] = {DATA_SIZE};
+				int tags[] = {TAG};
+				void* handles[] = {mhandles[i]};
+				OFINCCLCHECK(post_recv(ext_net, rComm, 1, recv_bufs, sizes, tags, handles, &requests[i]));
 			}
 		}
-	}
 
-	/** Close communicators with inflight requests **/
-	for (int idx = 0; idx < NUM_REQUESTS; idx++) {
-		if (rank == 0) {
-			OFINCCLCHECKGOTO(extNet->deregMr((void *)sComm, mhandle[idx]), res,
-						exit);
-		} else {
-			OFINCCLCHECKGOTO(extNet->deregMr((void *)rComm, mhandle[idx]), res,
-						exit);
+		// Close communicators with inflight requests (this is the test)
+		for (int i = 0; i < NUM_REQUESTS; i++) {
+			void* comm = (rank == 0) ? static_cast<void*>(sComm) : static_cast<void*>(rComm);
+			OFINCCLCHECK(deregister_memory(ext_net, comm, mhandles[i]));
```

Comment:
The comment says close communicators but here the memory is deregistered for an inflight request. Is this intentional? it might cause data aborts and unpredictable behavior. I don't think we are tracking the active requests in deregister call inside plugin.